src/CacheManager.Microsoft.Extensions.Caching.Memory/MemoryCacheHandle`1.cs

```diff
@@ -152,10 +173,7 @@ protected override bool AddInternalPrepared(CacheItem<TCacheValue> item)
             var options = GetOptions(item);
             _cache.Set(key, item, options);
 
-            if (item.Region != null)
-            {
-                _cache.RegisterChild(item.Region, key);
-            }
+            _cache.RegisterChild(item.Region ?? GetType().Name, key);
```

Comment:
If I recall correctly, I was having the region hashes show up in the key search (when it's metadata not intended data) which could result in the region hash being removed when the region wasn't empty. So I was using the type as a default region.

That might have been an artifact of my test cases covering both region and region-less cases though.